lib/internal/url.js

```diff
@@ -814,7 +831,33 @@ class URL {
     return `${constructor.name} ${inspect(obj, opts)}`;
   }
 
-  #updateContext(href) {
+  #getSearchFromContext() {
+    if (!this.#context.hasSearch) return '';
+    let endsAt = this.#context.href.length;
+    if (this.#context.hasHash) endsAt = this.#context.hash_start;
+    if (endsAt - this.#context.search_start <= 1) return '';
+    return StringPrototypeSlice(this.#context.href, this.#context.search_start, endsAt);
+  }
+
+  #getSearchFromParams() {
+    if (!this.#searchParams?.size) return '';
+    return `?${this.#searchParams}`;
+  }
+
+  #checkSearchParams() {
```

Comment:
I recommend something that starts with `ensure` since this not only checks but also updates it if required.

```suggestion
  #ensureSearchParamsValidity() {
```